src/main/java/frc/robot/commands/testing/TestCommand.java

```diff
@@ -0,0 +1,207 @@
+          shooterSubsystem.prepareToShoot(RotationsPerSecond.of(40));
+        }
+        if (shooterSubsystem.isReadyToShoot()) {
+          shooterSubsystem.stop();
+          timer.start();
+          hasStopped = true;
+        }
+        if (timer.hasElapsed(3)) {
+          teststate++;
+          timer.stop();
+          timer.reset();
+          hasStopped = false;
+        }
+        break;
+
+      case 3:
+        if (!hasStopped) {
+          shooterSubsystem.prepareToShoot(RotationsPerSecond.of(-40));
+        }
+        if (shooterSubsystem.isReadyToShoot()) {
+
+          shooterSubsystem.stop();
+          timer.start();
+          hasStopped = true;
+        }
+        if (timer.hasElapsed(3)) {
+          teststate++;
+          timer.stop();
+          timer.reset();
+          hasStopped = false;
+        }
+        break;
+
+      case 4:
+        if (!hasStopped) {
+          turretSubsystem.shoot();
+          timer.start();
+        }
+        if (timer.hasElapsed(2) && !hasStopped) {
+          turretSubsystem.stop();
+          hasStopped = true;
+        }
+        if (timer.hasElapsed(3)) {
+          teststate++;
+          timer.stop();
+          timer.reset();
+          hasStopped = false;
+        }
+        break;
+
+      case 5:
+        if (!hasStopped) {
+          turretSubsystem.eject();
+          timer.start();
+        }
+        if (timer.hasElapsed(2) && !hasStopped) {
+          turretSubsystem.stop();
+          hasStopped = true;
+        }
+        if (timer.hasElapsed(3)) {
+          teststate++;
+          timer.stop();
+          timer.reset();
+          hasStopped = false;
+        }
+        break;
+
+      case 6:
+        if (!hasStopped) {
+          turretSubsystem.moveToYawPosition(Degrees.of(-160));
+        }
+        if (turretSubsystem.isAtYawTarget()) {
+          turretSubsystem.prepareToIntake();
+          timer.start();
+          hasStopped = true;
+        }
+        if (timer.hasElapsed(3)) {
+          teststate++;
+          timer.stop();
+          timer.reset();
+          hasStopped = false;
+        }
+        break;
+
+      case 7:
+        if (!hasStopped) {
+          turretSubsystem.moveToPitchPosition(Degrees.of(10));
+        }
+        if (turretSubsystem.isAtPitchTarget()) {
+          turretSubsystem.prepareToIntake();
+          timer.start();
+          hasStopped = true;
+        }
+        if (timer.hasElapsed(3)) {
+          teststate++;
+          timer.stop();
+          timer.reset();
+          hasStopped = false;
+        }
+        break;
+    }
+  }
+
+  public int getTestState() {
+    return teststate;
+  }
+
+  public boolean getHasStopped() {
+    return hasStopped;
+  }
+
+  @Override
+  public boolean isFinished() {
+    return !RobotState.isTest();
+  }
+
+  @Override
+  public void end(boolean interrupted) {
+    shooterSubsystem.stop();
+    turretSubsystem.stop();
+    intakeSubsystem.stop();
+    turretSubsystem.prepareToIntake();
```

Comment:
Why `prepareToIntake()` at the end? Usually you would just want to stop the subsystems when you're done with them.

Reply:
No idea what I was thinking when I wrote that, but its fixed now.